development/components/database/prestashopcollection.md

```diff
@@ -0,0 +1,240 @@
+| like (or LIKE) | LIKE | Scalar |
+| notlike (or NOTLIKE) | NOT LIKE | Scalar |
+| regexp (or REGEXP) | REGEXP | Scalar |
+| notregexp (or NOTREGEXP) | NOT REGEXP | Scalar |
+
+### Examples of usage
+
+```php
+$productCollection = new PrestaShopCollection('Product');
+$productCollection->where('on_sale', '=', true); // find products on sale
+```
+
+```php
+$productCollection = new PrestaShopCollection('Product');
+$productCollection->where('reference', 'LIKE', 'REF-%'); // find products with reference beginning by "REF-"
+```
+
+### Having method
+
+Another method is available, `having()`, which calls `where()` with the parameter `$method` set to `having`. 
+
+```php
+public function having($field, $operator, $value)
+```
+
+## Joining to associated entities (join)
+
+When building the Collection, you may need to join with other ObjectModel entities. 
+You can join on associations that are declared in the $definition of the ObjectModel. 
+
+```php
+public function join($association, $on = '', $type = null)
+```
+
+Join `manufacturer` to your `Product` collection: 
+
+```php
+$productCollection = new PrestaShopCollection('Product');
+$productCollection->join('manufacturer');
+```
+
+You can join on a different field by specifying the field name as a second parameter: 
+
+```php
+$productCollection = new PrestaShopCollection('Product');
+$productCollection->join('categories', 'id_category');
+```
+
+By default, a `LEFT JOIN` will be used. `INNER JOIN` or `LEFT OUTER JOIN` are also available using the third parameter:
+
+```php
+$productCollection->join('categories', 'id_category', PrestaShopCollection::LEFT_JOIN);
+$productCollection->join('categories', 'id_category', PrestaShopCollection::INNER_JOIN);
+$productCollection->join('categories', 'id_category', PrestaShopCollection::LEFT_OUTER_JOIN);
+```
+
+### Using where with join
+
+One of the interest of joining other ObjectModel entities to your collection, is the possibility to filter on the external entity with `where()`. 
+
+```php
+$productCollection = (new PrestaShopCollection('Product'))
+    ->join('manufacturer')
+    ->where('manufacturer.name', '=', 'Manufacturer AAA');
+```
+
+## Ordering a Collection (order by)
+
+To order your collection, use the `orderBy()` method. 
+
+```php
+public function orderBy($field, $order = 'asc')
+```
+
+Ordering can be done in ascending or descending direction, with `asc` or `desc` `$order` parameter.
+
+```php
+$productCollection = (new PrestaShopCollection('Product'))
+    ->orderBy('reference', 'desc');
+```
+
+## Grouping (group by)
+
+To group your collection items on a field, use the `groupBy()` method. 
+
+```php
+public function groupBy($field)
+```
+
+```php
+$productCollection = (new PrestaShopCollection('Product'))
+    ->groupBy('id_supplier');
+```
+
+## Paginating a Collection (offset / limit)
+
+When fetching large quantities of items, you may need to paginate your collection, with SQL `OFFSET / LIMIT` equivalent mecanismes. 
```

Comment:
```suggestion
If you need to retrieve a large number of items, it may be necessary to use pagination techniques such as SQL's `OFFSET/LIMIT`. This will help you manage your collection more efficiently.
```